src/factory/CirclesBackingFactory.sol

```diff
@@ -0,0 +1,365 @@
+        releaseTimestamp = timestamp;
+    }
+
+    /// @notice Method sets supported status for backing asset.
+    function setSupportedBackingAssetStatus(address backingAsset, bool status) external onlyAdmin {
+        supportedBackingAssets[backingAsset] = status;
+    }
+
+    // Backing logic
+
+    /// @dev Required upfront approval of this contract for USDC.e
+    /// @dev Is called inside onERC1155Received callback by Hub call Circles 1155 transferFrom.
+    function startBacking(address backer, address backingAsset, address stableCRCAddress, uint256 stableCRCAmount)
+        internal
+    {
+        if (!supportedBackingAssets[backingAsset]) revert UnsupportedBackingAsset(backingAsset);
+
+        address instance = deployCirclesBacking(backer);
+
+        // transfer USDC.e
+        IERC20(USDC).transferFrom(backer, instance, TRADE_AMOUNT);
+        // transfer stable circles
+        IERC20(stableCRCAddress).transfer(instance, stableCRCAmount);
+
+        // create order
+        (, bytes32 appData) = getAppData(instance);
+        // Generate order UID using the "getUid" contract
+        (bytes32 orderDigest,) = GET_UID_CONTRACT.getUid(
+            USDC, // sellToken
+            backingAsset, // buyToken
+            instance, // receiver
+            TRADE_AMOUNT, // sellAmount
+            1, // buyAmount: Determined by off-chain logic or Cowswap solvers
+            VALID_TO, // order expiry
+            appData, // appData hash
+            0, // FeeAmount
+            true, // IsSell
+            false // PartiallyFillable
+        );
+        // Construct the order UID
+        bytes memory orderUid = abi.encodePacked(orderDigest, instance, uint32(VALID_TO));
+        // Initiate backing
+        CirclesBacking(instance).initiateBacking(
+            backer, backingAsset, stableCRCAddress, orderUid, USDC, TRADE_AMOUNT, stableCRCAmount
+        );
+        emit CirclesBackingInitiated(backer, instance, backingAsset, stableCRCAddress);
+    }
+
+    // LBP logic
+
+    /// @notice Creates LBP with underlying assets: `backingAssetAmount` backingAsset(`backingAsset`) and `CRC_AMOUNT` InflationaryCircles(`personalCRC`).
+    /// @param personalCRC Address of InflationaryCircles (stable ERC20) used as underlying asset in lbp.
+    /// @param backingAsset Address of backing asset used as underlying asset in lbp.
+    /// @param backingAssetAmount Amount of backing asset used in lbp.
+    function createLBP(address personalCRC, uint256 personalCRCAmount, address backingAsset, uint256 backingAssetAmount)
+        external
+        returns (address lbp, bytes32 poolId, IVault.JoinPoolRequest memory request, address vault)
+    {
+        address backer = backerOf[msg.sender];
+        if (backer == address(0)) revert OnlyCirclesBacking();
+
+        // prepare inputs
+        IERC20[] memory tokens = new IERC20[](2);
+        bool tokenZero = personalCRC < backingAsset;
```

Comment:
why is this important to do? would it not be easier to always have CRC be "the same side" in the LBPs?

Reply:
tokens must be sorted in ascending order in balancer and uniswap pools. the reason behind this is the price math, we need to know which ratio (e.g.,price = token1/token0) pool price is representing, because there can be different pools (based on different fees) with the same underlying tokens.